### PR TITLE
Make TS types easier to extend with strong typing

### PR DESCRIPTION
**Pull request by chase**
This is mostly backwards compatible, but it should make our code elsewhere not require a bunch of type coercion to get type safety and strongly typed APIs for postUnoCommand and the like.

### CODE REVIEW

File: src/electron/npm/libreoffice.d.ts [hunk 1]
```diff
@@ -91,49 +92,81 @@ declare namespace LibreOffice {
...
+
+  type CommandValueResult<R = { [name: string]: any }> = {
+    commandValues: R;
   };
 
-  type DocumentEventHandler<Event extends keyof DocumentEvents> = (
-    arg: DocumentEvents[Event]
-  ) => void;
+  interface GetCommands {
+    '.uno:PageColor': string;
```
Comment (whutchinson98) [inline src/electron/npm/libreoffice.d.ts:120]: I think we also need `.uno:GetOutline` here

**Comment by chase** [comment]
Also worth noting I made the necessary changes to monorepo to adopt these things, so nobody needs to work on that.

File: src/electron/npm/libreoffice.d.ts (hunk 1 above)
Comment (chase) [reply src/electron/npm/libreoffice.d.ts:120]: The point of this PR is that the actually exposed interfaces are generic, that's effectively a placeholder.

**Comment by whutchinson98** [comment]
> Also worth noting I made the necessary changes to monorepo to adopt these things, so nobody needs to work on that.

I was just typing and going to ask that very question. Thanks for calling it out.

File: src/electron/npm/libreoffice.d.ts (hunk 1 above)
Comment (chase) [reply src/electron/npm/libreoffice.d.ts:120]: Like this:
``` ts
import type { UNO, UnoCommands } from "@coparse/editor-uno-interface";

export interface DocumentEvents extends LibreOffice.DocumentEvents {}
export type DocumentClient = LibreOffice.DocumentClient<DocumentEvents, UNO, UnoCommands>
```

Comment (whutchinson98) [reply src/electron/npm/libreoffice.d.ts:120]: Oh ok. Sorry I misunderstood with that being there. Cool beans I'll approve! Looks like lots of fancy typescripting :D

**Review by whutchinson98 — APPROVED** [review]

**Pull request merged**